ftw/news/tests/test_news_listing_block.py

```diff
@@ -264,3 +265,34 @@ def test_has_image_class_is_not_set_on_show_lead_image(self, browser):
             map(lambda news_item: news_item.attrib['class'],
                 browser.css('.news-item .body'))
         )
+
+    @browsing
+    def test_news_listing_block_on_homepage(self, browser):
+        applyProfile(self.portal, 'ftw.news:show-on-homepage')
+
+        newsfolder = create(Builder('news folder'))
+
+        create(Builder('news')
+               .within(newsfolder)
+               .titled(u'News On Homepage')
+               .having(show_on_homepage=True)
+               .having(news_date=datetime(2011, 01, 02, 15, 0, 0)))
+
+        create(Builder('news')
+               .within(newsfolder)
+               .having(show_on_homepage=False)
+               .titled(u'News Not On Homepage')
+               .having(news_date=datetime(2011, 01, 02, 15, 0, 0)))
+
+        create(Builder('news listing block')
+               .within(self.portal)
+               .having(news_on_homepage=True))
+
+        browser.login()
+        browser.visit(self.portal, view='simplelayout-view')
+        self.assertEqual(
+            1,
```

Comment:
Could you try asserting the title here?
With this assertion it would be possible that we only see `News Not On Homepage`, which would be wrong 😉